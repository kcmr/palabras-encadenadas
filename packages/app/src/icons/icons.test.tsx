import renderer from 'react-test-renderer'
import * as icons from '.'
import type { ElementType } from 'react'

describe('Icons', () => {
  Object.keys(icons).forEach((name) => {
    it(`<${name}> renders the expected svg`, () => {
      const Icon = icons[name as keyof typeof icons] as ElementType
      const output = renderer.create(<Icon />).toJSON()
      expect(output).toMatchSnapshot()
    })
  })
})
