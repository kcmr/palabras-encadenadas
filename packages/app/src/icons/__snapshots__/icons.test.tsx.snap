// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icons <AlarmIcon> renders the expected svg 1`] = `
<svg
  fill="none"
  focusable={false}
  height="100%"
  viewBox="0 0 24 24"
  xmlns="http://www.w3.org/2000/svg"
>
  <circle
    cx="12"
    cy="13"
    r="9"
    stroke="currentColor"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
  />
  <path
    d="M15.5 9.5L12 13"
    stroke="currentColor"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
  />
  <path
    d="M19 19L20 22"
    stroke="currentColor"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
  />
  <path
    d="M5 19L4 22"
    stroke="currentColor"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
  />
  <path
    d="M2 5L5 2"
    stroke="currentColor"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
  />
  <path
    d="M19 2L22 5"
    stroke="currentColor"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
  />
  <path
    d="M12 4V2"
    stroke="currentColor"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
  />
</svg>
`;

exports[`Icons <AlertIcon> renders the expected svg 1`] = `
<svg
  fill="none"
  focusable={false}
  viewBox="0 0 24 24"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M12 9V14"
    stroke="currentColor"
    strokeLinecap="round"
    strokeWidth="2"
  />
  <path
    d="M12 17.5V18"
    stroke="currentColor"
    strokeLinecap="round"
    strokeWidth="2"
  />
  <path
    d="M2.23231 19.0159L10.3489 3.05158C11.0617 1.64947 12.9383 1.64947 13.6512 3.05158L21.7677 19.0159C22.4513 20.3606 21.5442 22 20.1165 22H3.88346C2.45582 22 1.54868 20.3606 2.23231 19.0159Z"
    stroke="currentColor"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
  />
</svg>
`;

exports[`Icons <ArrowIcon> renders the expected svg 1`] = `
<svg
  fill="none"
  focusable={false}
  viewBox="0 0 24 24"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M8 5L3 10L8 15"
    stroke="currentColor"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
  />
  <path
    d="M3 10H11C16.5228 10 21 14.4772 21 20V21"
    stroke="currentColor"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
  />
</svg>
`;

exports[`Icons <CrossIcon> renders the expected svg 1`] = `
<svg
  fill="none"
  focusable={false}
  viewBox="0 0 24 24"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M20 20L4 4.00003M20 4L4.00002 20"
    stroke="currentColor"
    strokeLinecap="round"
    strokeWidth={2}
  />
</svg>
`;

exports[`Icons <FacebookIcon> renders the expected svg 1`] = `
<svg
  fill="none"
  viewBox="0 0 32 32"
  xmlns="http://www.w3.org/2000/svg"
>
  <circle
    cx="16"
    cy="16"
    fill="url(#paint0_linear_87_7208)"
    r="14"
  />
  <path
    d="M21.2137 20.2816L21.8356 16.3301H17.9452V13.767C17.9452 12.6857 18.4877 11.6311 20.2302 11.6311H22V8.26699C22 8.26699 20.3945 8 18.8603 8C15.6548 8 13.5617 9.89294 13.5617 13.3184V16.3301H10V20.2816H13.5617V29.8345C14.2767 29.944 15.0082 30 15.7534 30C16.4986 30 17.2302 29.944 17.9452 29.8345V20.2816H21.2137Z"
    fill="white"
  />
  <defs>
    <linearGradient
      gradientUnits="userSpaceOnUse"
      id="paint0_linear_87_7208"
      x1="16"
      x2="16"
      y1="2"
      y2="29.917"
    >
      <stop
        stopColor="#18ACFE"
      />
      <stop
        offset="1"
        stopColor="#0163E0"
      />
    </linearGradient>
  </defs>
</svg>
`;

exports[`Icons <InfoIcon> renders the expected svg 1`] = `
<svg
  fill="none"
  focusable={false}
  viewBox="0 0 24 24"
  xmlns="http://www.w3.org/2000/svg"
>
  <circle
    cx="12"
    cy="12"
    r="10"
    stroke="currentColor"
    strokeWidth="2"
  />
  <path
    d="M12 7H12.01"
    stroke="currentColor"
    strokeLinecap="round"
    strokeWidth="2"
  />
  <path
    d="M10 11H12V16"
    stroke="currentColor"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
  />
  <path
    d="M10 16H14"
    stroke="currentColor"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
  />
</svg>
`;

exports[`Icons <PlayIcon> renders the expected svg 1`] = `
<svg
  fill="none"
  viewBox="0 0 24 24"
  xmlns="http://www.w3.org/2000/svg"
>
  <circle
    cx="12"
    cy="12"
    r="10"
    stroke="currentColor"
    strokeWidth={2}
  />
  <path
    d="M16 12L10 8V16L16 12Z"
    stroke="currentColor"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth={2}
  />
</svg>
`;

exports[`Icons <SadIcon> renders the expected svg 1`] = `
<svg
  fill="none"
  focusable={false}
  viewBox="0 0 24 24"
  xmlns="http://www.w3.org/2000/svg"
>
  <circle
    cx="12"
    cy="12"
    r="10"
    stroke="currentColor"
    strokeLinecap="round"
    strokeWidth="2"
  />
  <path
    d="M8 9.05001V8.95001"
    stroke="currentColor"
    strokeLinecap="round"
    strokeWidth="2"
  />
  <path
    d="M16 9.05001V8.95001"
    stroke="currentColor"
    strokeLinecap="round"
    strokeWidth="2"
  />
  <path
    d="M16 16C15.5 14.5 14.2091 13 12 13C9.79086 13 8.5 14.5 8 16"
    stroke="currentColor"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
  />
</svg>
`;

exports[`Icons <ShareIcon> renders the expected svg 1`] = `
<svg
  fill="none"
  focusable={false}
  viewBox="0 0 24 24"
  xmlns="http://www.w3.org/2000/svg"
>
  <circle
    cx="18"
    cy="5"
    r="3"
    stroke="currentColor"
    strokeWidth={2}
  />
  <circle
    cx="18"
    cy="19"
    r="3"
    stroke="currentColor"
    strokeWidth={2}
  />
  <circle
    cx="6"
    cy="12"
    r="3"
    stroke="currentColor"
    strokeWidth={2}
  />
  <path
    d="M15.408 6.51199L8.59436 10.4866M15.408 17.488L8.59436 13.5134"
    stroke="currentColor"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth={2}
  />
</svg>
`;

exports[`Icons <TwitterIcon> renders the expected svg 1`] = `
<svg
  fill="none"
  viewBox="0 0 48 48"
  xmlns="http://www.w3.org/2000/svg"
>
  <circle
    cx="24"
    cy="24"
    fill="#1DA1F2"
    r="20"
  />
  <path
    clipRule="evenodd"
    d="M36 16.3086C35.1177 16.7006 34.1681 16.9646 33.1722 17.0838C34.1889 16.4742 34.9697 15.5095 35.3368 14.36C34.3865 14.9247 33.3314 15.3335 32.2107 15.5551C31.3123 14.5984 30.0316 14 28.6165 14C25.8975 14 23.6928 16.2047 23.6928 18.9237C23.6928 19.3092 23.7368 19.6852 23.8208 20.046C19.7283 19.8412 16.1005 17.8805 13.6719 14.9015C13.2479 15.6287 13.0055 16.4742 13.0055 17.3766C13.0055 19.0845 13.8735 20.5916 15.1958 21.4747C14.3878 21.4491 13.6295 21.2275 12.9647 20.8587V20.9203C12.9647 23.3066 14.663 25.296 16.9141 25.7496C16.5013 25.8616 16.0661 25.9224 15.6174 25.9224C15.2998 25.9224 14.991 25.8912 14.6902 25.8336C15.3166 27.7895 17.1357 29.2134 19.2899 29.2534C17.6052 30.5733 15.4822 31.3612 13.1751 31.3612C12.7767 31.3612 12.3848 31.338 12 31.2916C14.1791 32.6884 16.7669 33.5043 19.5475 33.5043C28.6037 33.5043 33.5562 26.0016 33.5562 19.4956C33.5562 19.282 33.5522 19.0693 33.5418 18.8589C34.5049 18.1629 35.34 17.2958 36 16.3086Z"
    fill="white"
    fillRule="evenodd"
  />
</svg>
`;

exports[`Icons <VictoryIcon> renders the expected svg 1`] = `
<svg
  fill="none"
  focusable={false}
  viewBox="0 0 24 24"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M12 11L11 3.72792C11 3.26184 11.1851 2.81485 11.5147 2.48528C12.5387 1.46129 14.2923 2.04581 14.4971 3.47939L16 10L17.5081 3.6723C17.7769 2.75853 18.7501 2.25003 19.6537 2.55123C20.4313 2.81044 20.9088 3.59248 20.7842 4.40265L19 16"
    stroke="currentColor"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth={2}
  />
  <path
    d="M19 16C18.4643 20 15.6421 22 11.5 22C7.35786 22 4 20 4 16"
    stroke="currentColor"
    strokeWidth={2}
  />
  <path
    d="M4 16V12C4 10.8954 4.89543 10 6 10C7.10457 10 8 10.8954 8 12M12 13V11C12 9.89543 11.1046 9 10 9C8.89543 9 8 9.89543 8 11V15"
    stroke="currentColor"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth={2}
  />
  <path
    d="M13.6923 17H11C9.89543 17 9 16.1046 9 15C9 13.8954 9.89543 13 11 13L15 13C17.2091 13 19.5 15 18.5 18"
    stroke="currentColor"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth={2}
  />
</svg>
`;

exports[`Icons <WhatsappIcon> renders the expected svg 1`] = `
<svg
  fill="none"
  viewBox="0 0 32 32"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    clipRule="evenodd"
    d="M16 31C23.732 31 30 24.732 30 17C30 9.26801 23.732 3 16 3C8.26801 3 2 9.26801 2 17C2 19.5109 2.661 21.8674 3.81847 23.905L2 31L9.31486 29.3038C11.3014 30.3854 13.5789 31 16 31ZM16 28.8462C22.5425 28.8462 27.8462 23.5425 27.8462 17C27.8462 10.4576 22.5425 5.15385 16 5.15385C9.45755 5.15385 4.15385 10.4576 4.15385 17C4.15385 19.5261 4.9445 21.8675 6.29184 23.7902L5.23077 27.7692L9.27993 26.7569C11.1894 28.0746 13.5046 28.8462 16 28.8462Z"
    fill="#BFC8D0"
    fillRule="evenodd"
  />
  <path
    d="M28 16C28 22.6274 22.6274 28 16 28C13.4722 28 11.1269 27.2184 9.19266 25.8837L5.09091 26.9091L6.16576 22.8784C4.80092 20.9307 4 18.5589 4 16C4 9.37258 9.37258 4 16 4C22.6274 4 28 9.37258 28 16Z"
    fill="url(#paint0_linear_87_7264)"
  />
  <path
    clipRule="evenodd"
    d="M16 30C23.732 30 30 23.732 30 16C30 8.26801 23.732 2 16 2C8.26801 2 2 8.26801 2 16C2 18.5109 2.661 20.8674 3.81847 22.905L2 30L9.31486 28.3038C11.3014 29.3854 13.5789 30 16 30ZM16 27.8462C22.5425 27.8462 27.8462 22.5425 27.8462 16C27.8462 9.45755 22.5425 4.15385 16 4.15385C9.45755 4.15385 4.15385 9.45755 4.15385 16C4.15385 18.5261 4.9445 20.8675 6.29184 22.7902L5.23077 26.7692L9.27993 25.7569C11.1894 27.0746 13.5046 27.8462 16 27.8462Z"
    fill="white"
    fillRule="evenodd"
  />
  <path
    d="M12.5 9.49989C12.1672 8.83131 11.6565 8.8905 11.1407 8.8905C10.2188 8.8905 8.78125 9.99478 8.78125 12.05C8.78125 13.7343 9.52345 15.578 12.0244 18.3361C14.438 20.9979 17.6094 22.3748 20.2422 22.3279C22.875 22.2811 23.4167 20.0154 23.4167 19.2503C23.4167 18.9112 23.2062 18.742 23.0613 18.696C22.1641 18.2654 20.5093 17.4631 20.1328 17.3124C19.7563 17.1617 19.5597 17.3656 19.4375 17.4765C19.0961 17.8018 18.4193 18.7608 18.1875 18.9765C17.9558 19.1922 17.6103 19.083 17.4665 19.0015C16.9374 18.7892 15.5029 18.1511 14.3595 17.0426C12.9453 15.6718 12.8623 15.2001 12.5959 14.7803C12.3828 14.4444 12.5392 14.2384 12.6172 14.1483C12.9219 13.7968 13.3426 13.254 13.5313 12.9843C13.7199 12.7145 13.5702 12.305 13.4803 12.05C13.0938 10.953 12.7663 10.0347 12.5 9.49989Z"
    fill="white"
  />
  <defs>
    <linearGradient
      gradientUnits="userSpaceOnUse"
      id="paint0_linear_87_7264"
      x1="26.5"
      x2="4"
      y1="7"
      y2="28"
    >
      <stop
        stopColor="#5BD066"
      />
      <stop
        offset="1"
        stopColor="#27B43E"
      />
    </linearGradient>
  </defs>
</svg>
`;
